climpred/prediction.py

```diff
@@ -257,11 +260,22 @@ def compute_hindcast(
     for i in forecast.lead.values:
         if max_dof:
             forecast, reference = reduce_time_series(forecast, reference, i)
-        # take lead year i timeseries and convert to real time
+
+        # take lead  i timeseries and convert to real time based on temporal
+        # resolution of lead (lead_res)
+        offset_args_dict = get_lead_pdoffset_args(forecast['lead'].attrs['units'], i)
         a = forecast.sel(lead=i).drop('lead')
-        a['time'] = [int(t + i) for t in a.time.values]
-        # take real time reference of real time forecast years
+        a['time'] = pd.to_datetime(
+            a['time'].dt.strftime('%Y%m%d 00:00')
+        ) + pd.DateOffset(**offset_args_dict)
+
+        # take real time reference of real time forecast dates
         b = reference.sel(time=a.time.values)
+
+        # Rechunk to single chunk over time to avoid error
+        a = a.chunk({'time': -1})
+        b = b.chunk({'time': -1})
```

Comment:
This should be deleted. I think (1) it imposes dask on a user when they might not need or want it. And (2)  some metric/dim operations don't need time fully chunked. The error is pretty explanatory when it comes through for them to fix.

Reply:
or maybe check whether a and b are dask arrays and only if time is chunked then do that

Reply:
But there are some cases where you don't need/want time as a full chunk. E.g. if you were doing pattern correlation. I think the user should handle the chunking on their own. Since if they need to rechunk `time` to a full chunk the error message is very clear.

Reply:
I agree that its probably best not to assume dask or how the data should be chunked, but leave this to the user. I removed this and tested in my notebooks to have the user do the re-chunking external to climpred.  If the user does not re-chunk, the error message is very clear.

Reply:
I encountered this issue while testing a notebook on your branch. It can be frustrating. I think you could do:

```python
if dask.is_dask_collection(a):
    a = a.chunk({'time': -1})
    ...
```

So you only force chunk if it's already chunked ( a dask collection )